dhcp/pxe.go

```diff
@@ -126,13 +126,24 @@ func SetupPXE(ctx context.Context, rep, req *dhcp4.Packet) bool {
 		Options 64-127 are "boot server specific" so that's why we put traceparent
 		propagation in opt43/slot69.
 	*/
-
+	cur, ok := rep.GetOption(dhcp4.OptionVendorSpecific)
+	if !ok {
+		c := make(dhcp4.OptionMap, 0)
+		cur = c.Serialize()
+	}
 	pxeVendorOptions := dhcp4.OptionMap{
 		6:  []byte{0x8}, // PXE_DISCOVERY_CONTROL: Attempt to tell PXE to boot faster.
 		69: binaryTpFromContext(ctx),
-	}.Serialize()
+	}
+	err := pxeVendorOptions.Deserialize(cur, nil) // merge the current option 43 with pxeVendorOptions.
+	if err != nil {
+		pxeVendorOptions = dhcp4.OptionMap{
+			6:  []byte{0x8}, // PXE_DISCOVERY_CONTROL: Attempt to tell PXE to boot faster.
+			69: binaryTpFromContext(ctx),
+		}
+	}
 
-	rep.SetOption(dhcp4.OptionVendorSpecific, pxeVendorOptions)
+	rep.SetOption(dhcp4.OptionVendorSpecific, pxeVendorOptions.Serialize())
```

Comment:
similarly this seems like less/easier-to-follow code:
```go
om := make(dhcp4.OptionMap, 0)

cur, ok := rep.GetOption(dhcp4.OptionVendorSpecific)
if ok {
	om.Deserialize(cur, nil)
}

om[6] = []byte{0x8} // PXE_DISCOVERY_CONTROL: Attempt to tell PXE to boot faster.
om[69] = binaryTpFromContext(ctx)

rep.SetOption(dhcp4.OptionVendorSpecific, om.Serialize())
```

Its easier to work with `OptionMap`s (plain old go maps) than `[]byte`s through the `Serialize`/`Deserialize` methods imo.